hazelcast/src/test/java/com/hazelcast/client/queue/QueueResetAgeStatisticsTest.java

```diff
@@ -0,0 +1,101 @@
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.MCResetQueueAgeStatisticsCodec;
+import com.hazelcast.client.impl.spi.impl.ClientInvocation;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.collection.IQueue;
+import com.hazelcast.collection.LocalQueueStats;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class QueueResetAgeStatisticsTest
+        extends HazelcastTestSupport {
+
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+    private HazelcastClientInstanceImpl client;
+    private HazelcastInstance member;
+
+    @Before
+    public void setup() {
+        member = hazelcastFactory.newHazelcastInstance();
+        client = ((HazelcastClientProxy) hazelcastFactory.newHazelcastClient()).client;
+    }
+
+    @Test
+    public void testAgeStatsReset()
+            throws ExecutionException, InterruptedException {
+        IQueue<String> myQueue = client.getQueue("my-queue");
+        assertEquals(Long.MAX_VALUE, member.getQueue("my-queue").getLocalQueueStats().getMinAge());
+        myQueue.add("item-1");
+        myQueue.add("item-2");
+        Thread.sleep(50);
+        myQueue.take();
+        myQueue.take();
+
+        LocalQueueStats stats = member.getQueue("my-queue").getLocalQueueStats();
+        assertTrue(stats.getMaxAge() > 0);
+        assertTrue(stats.getMinAge() > 0);
+        assertTrue(stats.getAverageAge() > 0);
+
+        ClientMessage clientMessage = MCResetQueueAgeStatisticsCodec.encodeRequest("my-queue");
+        ClientInvocation invocation = new ClientInvocation(client, clientMessage, "my-queue");
+        invocation.invoke().get();
+        LocalQueueStats statsAfterReset = member.getQueue("my-queue").getLocalQueueStats();
+        assertEquals(0, statsAfterReset.getMaxAge());
+        assertEquals(Long.MAX_VALUE, statsAfterReset.getMinAge());
+        assertEquals(0, statsAfterReset.getAverageAge());
+
+        myQueue.add("item-3");
+        myQueue.add("item-4");
+        Thread.sleep(20);
+        myQueue.take();
+        Thread.sleep(100);
+        myQueue.take();
+
+        LocalQueueStats populatedStatsAfterReset = member.getQueue("my-queue").getLocalQueueStats();
+        assertTrue(statsAfterReset.getMinAge() + " is greater than 20", statsAfterReset.getMinAge() > 20);
```

Comment:
I see that this assertion failed a couple of times. What do you think about changing the bare sleeps above with `sleepAtLeastMillis(duration);` and changing the assertion from `>` to `>=`?

Reply:
Done, thanks. I hope it won't fail anymore (indeed the failures were probably caused by the strict `>` instead of `>=`).